frontend/pages/shelter/pet/add.tsx

```diff
@@ -0,0 +1,443 @@
+import { DeleteOutlined, EditOutlined, EyeOutlined } from "@ant-design/icons";
+import {
+	Breadcrumb,
+	Button,
+	DatePicker,
+	Input,
+	Radio,
+	RadioChangeEvent,
+	Select
+} from "antd";
+import { Content } from "antd/lib/layout/layout";
+import Title from "antd/lib/typography/Title";
+import { FurLength, PetData, Sex, Species, Status } from "common/enums";
+import ShelterLayout from "layouts/shelter/ShelterLayout";
+import moment from "moment";
+import { useRouter } from "next/dist/client/router";
+import React, { ReactNode, useEffect, useRef, useState } from "react";
+import styled from "styled-components";
+
+export default function EditPetDetails() {
+	const router = useRouter();
+	const { id } = router.query;
+	const [pet, setPet] = useState<PetData>({
+		key: "",
+		name: "",
+		images: [],
+		visible: true,
+		sex: Sex.MALE,
+		species: Species.CAT,
+		status: Status.HEALTHY,
+		acquired: new Date(),
+		breed: "",
+		furLength: FurLength.SHORT,
+		medicalIssues: [],
+		sterilised: "yes",
+		dateOfBirth: new Date(),
+		furColor: [],
+		toiletTrained: true,
+	});
+
+	const onRadioChange = (e: RadioChangeEvent, key: keyof Pick<PetData, "visible" | "toiletTrained" | "sex" | "sterilised" | "status">, isYesNo?: boolean) => {
+		if (!e.target.value) return;
+		const val = isYesNo ? e.target.value === "yes" ? true : false : e.target.value;
+		setPet((prev) => ({ ...prev, [key]: val }));
+	};
+
+	const onSelectChange = (value: string | string[], key: keyof Pick<PetData, "species" | "furLength" | "breed" | "medicalIssues">) => {
+		if (!value) return;
+		setPet((prev) => ({ ...prev, [key]: value }));
+	};
+
+	const onDateChange = (date: moment.Moment, key: keyof Pick<PetData, "acquired" | "dateOfBirth">) => {
+		if (!date) return;
+		setPet((prev) => ({ ...prev, [key]: date.toDate() }));
+	};
+
+	const onGalleryChange = (images: string[]) => {
```

Comment:
Getting an error due to this as it is passed in as a prop to `ImageGallery` and is then a dependency for `useEffect`

Reply:
actual error might be due to how the set of images is being updated repeatedly